### DISCUSSION TITLE
Support `AbortSignal` from the app layer

### DISCUSSION THREAD

**Discussion by fubhy** [wevm/viem#1239] [Idea / Feature Request]
It would be good if it was possible to pass an `AbortSignal` to anything that may trigger an rpc request (e.g. `simulateContract`, `readContract`, `estimateContractGas`). I recognize how that would require some custom handling for batch / multicall but it should be doable. 

`AbortSignal`s are already supported internally at the transport layer but they are not exposed to the app layer. Hence it's currently not possible to cancel requests from the app layer.

**Comment by au2001** [comment]
I would really like to see this implemented

A use case for this is React Hooks (e.g. wagmi's): each action performed in a `useEffect` should have a cleanup function
But there is currently no way to cancel requests made by viem AFAIK, so they are subject to memory leaks and race conditions
This Feature Request would be a nice solution and is the recommended one according to [the React docs](https://react.dev/reference/react/useEffect#fetching-data-with-effects) and [the linked article](https://maxrozen.com/race-conditions-fetching-data-react-with-useeffect)

Related issue in ethers.js: https://github.com/ethers-io/ethers.js/issues/4122

**Comment by fubhy** [comment]
Sounds like this will be implemented via https://github.com/wevm/viem/discussions/1701
